module_ctx.go

```diff
@@ -278,6 +282,7 @@ type moduleContext struct {
 	handledMissingDeps bool
 }
 
+// TODO: expain how this differs from logicModule.Name() : does this relativize/contextualize paths?
```

Comment:
See the comment on Module.Name().  logicModule.Name() is called exactly once, and returns what the module is initially know as.  moduleInfo.Name() returns the current name, which is initially the value of logicModule.Name(), but could be renamed by MutatorContext.Rename().